File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode MiddleNode(ListNode head) {
        //  https://leetcode.com/problems/middle-of-the-linked-list/solutions/4834666/best-approach-explained-with-gfg-image
        ListNode[] nodes = new ListNode[100];
        int index = 0;

        while(head != null){
            nodes[index++] = head;
            head = head.next;
        }

        return nodes[index / 2];
    }
}

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        vector<ListNode*> vec;
        while(head != NULL){
            vec.push_back(head);
            head = head->next;
        }
        return vec.size() % 2 == 1 ? vec[(vec.size() - 1) / 2] : vec[(vec.size() - 1) / 2 + 1];
    }
};